### ISSUE TITLE
Rule suggestion: Add rule for consistent STRING quotes for dialect that allow multiple quote types

### ISSUE THREAD

**Issue by dmohns** [sqlfluff/sqlfluff#3090]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

Some dialects allow STRINGs to be quoted using different kind of quotes. Most of the times teams seem to have a tendency towards one style of quotes and mights wish to specify this as a part of their SQLFluff rule set.

The rule could have a configuration option `string_quotes` that takes values `consistent`, `single_quotes` or `double_quotes`.

### Use case

From what I can see this affects `mysql` and `bigquery`. In both of which the following is a valid query

```sql
SELECT
  'abc' AS col1,
  "abc" AS col2,
  'abc' = "abc" AS col3
```

and could be linted to 

```sql
SELECT
  'abc' AS col1,
  'abc' AS col2,
  'abc' = 'abc' AS col3
```

in the case of `consistent` or `single_quotes`.

And linted to 

```sql
SELECT
  "abc" AS col1,
  "abc" AS col2,
  "abc" = "abc" AS col3
```

in the case of `double_quotes`.

### Dialect

bigquery, mysql

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
